regexp_test.go

```diff
@@ -601,26 +602,26 @@ func TestRegexpOverrideSpecies(t *testing.T) {
 	testScript1(SCRIPT, _undefined, t)
 }
 
-func TestRegexpSymbolMatchAllCallsIsRegexp(t *testing.T) {
-	// This is tc39's test/built-ins/RegExp/prototype/Symbol.matchAll/isregexp-this-throws.js
-	const SCRIPT = `
-	var a = new Object();
-	Object.defineProperty(a, Symbol.match, {
-		get: function() {
-			throw "passed";
-		}
-	});
-	try {
-		RegExp.prototype[Symbol.matchAll].call(a, '');
-		throw new Error("Expected error");
-	} catch(e) {
-		if (e !== "passed") {
-			throw e;
-		}
-	}
-	`
-	testScript1(SCRIPT, _undefined, t)
-}
+// func TestRegexpSymbolMatchAllCallsIsRegexp(t *testing.T) {
```

Comment:
Another two more tests this and the `TestRegexpLookbehindAssertion` are currently failing. Didn't get a chance to dig much into it yet.

Reply:
we added some specific regex fixes so I'm going to assume those need to be reapplied

Reply:
This (and the `native_test`) is now passing!

It happened by using this

```go
func (r *Runtime) toConstructor(v Value) func(args []Value, newTarget *Object) *Object {
	if ctor := r.toObject(v).self.assertConstructor(); ctor != nil {
		return ctor
	}

        // panic("Value is not a constructor") Before!
	panic(r.NewTypeError("Value is not a constructor"))
}
```